jwst/pipeline/calwebb_spec3.py

```diff
@@ -95,6 +98,16 @@ def process(self, input):
         self.combine_1d.save_results = self.save_results
         self.spectral_leak.suffix = 'x1d'
         self.spectral_leak.save_results = self.save_results
+        self.pixel_replace.suffix = 'pixel_replace'
+        self.pixel_replace.save_results = self.save_results
+        self.pixel_replace.output_use_model = True
+
+        # Overriding the Step.save_model method for the following steps.
+        # These steps save intermediate files, resulting in meta.filename
+        # being modified. This can affect the filenames of subsequent
+        # steps.
+        self.outlier_detection.save_model = invariant_filename(self.outlier_detection.save_model)
```

Comment:
Were the output file names for the `outlier_detection` step not being handled properly before?  How does this relate to the change that Brett recently implemented where the "asn_id" is added to the `outlier_detection` step output file names, in order to prevent name collisions in parallel processing?  I hope this doesn't somehow negate or otherwise react badly with those changes.

Reply:
This change was made by Jonathan to make the filenames be correct. As I understand it outlier detection was modifying the name and now pixel replacement took that name and added to it. We ended up with the asn_id being in the file multiple times.  Before it did not matter because outlier detection was the last step that wrote out names. But now it became an issue. Jonathan cleaned this up by adding this invariant_filename function